src/test/java/qna/domain/QuestionTest.java

```diff
@@ -1,6 +1,36 @@
 package qna.domain;
 
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import java.util.List;
+import org.junit.jupiter.api.Test;
+import qna.CannotDeleteException;
+
 public class QuestionTest {
     public static final Question Q1 = new Question("title1", "contents1").writeBy(UserTest.JAVAJIGI);
     public static final Question Q2 = new Question("title2", "contents2").writeBy(UserTest.SANJIGI);
+
+    @Test
+    void testDeleteSuccess() throws CannotDeleteException {
+        List<DeleteHistory> deleteHistories = Q1.deleteBy(UserTest.JAVAJIGI);
+
+        assertThat(Q1.isDeleted()).isTrue();
+        deleteHistories.forEach(deleteHistory -> assertThat(deleteHistory.isDeletedBy(Q1.getWriter())).isTrue());
+    }
+
+    @Test
+    void testDeleteFailByOtherUser() {
+        assertThatThrownBy(() -> Q1. deleteBy(UserTest.SANJIGI))
+            .isInstanceOf(CannotDeleteException.class)
+            .hasMessageContaining("질문을 삭제할 권한이 없습니다.");
+    }
+
+    @Test
+    void testDeleteFailByExistingOtherAnswers() {
+        Q1.addAnswer(AnswerTest.A2);
+        assertThatThrownBy(() -> Q1.deleteBy(UserTest.JAVAJIGI))
+            .isInstanceOf(CannotDeleteException.class)
+            .hasMessageContaining("다른 사람이 쓴 답변이 있어 삭제할 수 없습니다.");
+    }
```

Comment:
```suggestion
    @Test
    void testDeleteFailByExistingOtherAnswers() {
        Q1.addAnswer(AnswerTest.A2);
        assertThat(Q1.isDeleted()).isTrue();
        assertThatThrownBy(() -> Q1.deleteBy(UserTest.JAVAJIGI))
            .isInstanceOf(CannotDeleteException.class)
            .hasMessageContaining("다른 사람이 쓴 답변이 있어 삭제할 수 없습니다.");
    }
```

이렇게 테스트 코드를 수정한 후 클래스 단위, 또는 패키지 단위로 테스트를 수행해 보면 테스트가 통과하는 것을 알 수 있어요. 즉, `deleteBy()` 메서드를 호출하지 않았는데도 이미 삭제된 상태인 것이죠. 왜 이런 현상이 발생하는 걸까요? 이렇게 되어도 괜찮은 테스트일까요?